utils/model_runners/gpt_model_runner.py

```diff
@@ -38,12 +44,46 @@ def predict(self, prompt: str) -> Tuple[Optional[str], Optional[float]]:
             "presence_penalty": 0,
             "frequency_penalty": 0
         })
-        
+
         headers = {
             "Content-Type": "application/json",
             "Accept": "application/json",
             "Authorization": "Bearer " + self.config.api_key
         }
-        
+        stime = datetime.now(timezone.utc)
+
         response = requests.request("POST", self.url, headers=headers, data=payload)
-        return json.loads(response.text)["choices"][0]["message"]["content"], None
\ No newline at end of file
+        delta =  datetime.now(timezone.utc) - stime
+        processing_time = delta.total_seconds() * 1000
+
+
+        response = json.loads(response.text)
+        output = response["choices"][0]["message"]["content"]
+        input_token_count = int(response["usage"]["prompt_tokens"])        
+        output_token_count = int(response["usage"]["completion_tokens"])
+        
+        if self._reset and (self._metrics_folder is not None) and os.path.exists(self._metrics_folder + f"/{self._model_key}_usage.jsonl"):
+            os.remove(self._metrics_folder + f"/{self._model_key}_usage.jsonl")
+
+        sw = json.dumps({"input_tokens":input_token_count,"output_tokens":output_token_count, "processing_time":processing_time,
+                             "cost":calculate_usage_cost(model_id=self.config.model_id, input_tokens=input_token_count, output_tokens=output_token_count, inference_time_ms=processing_time)})
```

Comment:
Cost should be calculated outside of the ModelRunner, the runner should do the minimal logic of running the model and collecting the raw values (like latency and in/out tokens), cost should be calculate later on in the flow. Please remove it from the GPT/Bedrock runners.